File: contents/bubble_sort/code/r/bubblesort.R
bubble_sort <- function(array) {
  len_array <- length(array)
  for (i in 1:(len_array-1)) {
    for (j in 1:(len_array-i)) {
      if (array[j] > array[j+1]) {
        #'temp' will hold the current array[j] value because it will be changed
        temp <- array[j]
        array[j] <- array[j+1]
        array[j+1] <- temp
      }
    }
  }
  return(array)
}

main_function <- function() {
  #the sample function creates a vector/1d array with random integers
  #100 states the right endpoint of the closed interval, [1, 100]
  #15 states the amount of integers we want in our vector/1d array
  random_array <- sample(100, 15)
  print("Before sorting:")
  print(random_array)
  print("After sorting:")
  print(bubble_sort(random_array))
}

main_function()

File: contents/bubble_sort/code/python/bubblesort.py
import random


def bubble_sort(array):
    len_array = len(array)
    for i in range(len_array):
        for j in range(len_array - i - 1):
            if(array[j] > array[j+1]):
                array[j], array[j+1] = array[j+1], array[j] #swap elements in the list

def main():
    random_array = [random.randint(0, 1000) for _ in range(10)]
    print("Before Sorting  {}".format(random_array))
    bubble_sort(random_array)
    print("After Sorting  {}".format(random_array))

main()
